### PR TITLE
fix discrete_inputs & fix read coils bit * 8 & fix modbus decode coils

### PR DESCRIPTION
**Pull request by zhufeng88**

### CODE REVIEW

File: thingsboard_gateway/connectors/modbus/bytes_modbus_uplink_converter.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def convert(self, config, data):
                     word_endian_order = Endian.Little if word_order.upper() == "LITTLE" else Endian.Big
                     decoded_data = None
                     if not isinstance(response, ModbusIOException) and not isinstance(response, ExceptionResponse):
-                        if configuration["functionCode"] == 1:
+                        if configuration["functionCode"] in [1,2] :
```
Comment (samson0v) [inline thingsboard_gateway/connectors/modbus/bytes_modbus_uplink_converter.py:56]: Why did you change this function's number?

**Comment by zhufeng88** [comment]
Hi
function 2  response is like function 1 response ,is return bits,is not return register, so change function number.
________________________________
发件人: Vitalii Bidochka ***@***.***>
发送时间: 2022年6月28日 18:35
收件人: thingsboard/thingsboard-gateway ***@***.***>
抄送: Phoom ***@***.***>; Author ***@***.***>
主题: Re: [thingsboard/thingsboard-gateway] fix discrete_inputs & fix read coils bit * 8 & fix modbus decode coils (PR #860)


@samson0v commented on this pull request.

________________________________

In thingsboard_gateway/connectors/modbus/bytes_modbus_uplink_converter.py<https://github.com/thingsboard/thingsboard-gateway/pull/860#discussion_r908321355>:

> @@ -53,7 +53,7 @@ def convert(self, config, data):
                     word_endian_order = Endian.Little if word_order.upper() == "LITTLE" else Endian.Big
                     decoded_data = None
                     if not isinstance(response, ModbusIOException) and not isinstance(response, ExceptionResponse):
-                        if configuration["functionCode"] == 1:
+                        if configuration["functionCode"] in [1,2] :


Why did you change this function's number?

―

File: thingsboard_gateway/connectors/modbus/bytes_modbus_uplink_converter.py [hunk 2]
```diff
@@ -62,8 +62,9 @@ def convert(self, config, data):
                                 decoder = BinaryPayloadDecoder.fromCoils(coils, wordorder=word_endian_order)
                             assert decoder is not None
 
-                            decoded_data = decoder.decode_bits()[0]
-                        elif configuration["functionCode"] in [2, 3, 4]:
+                            #decoded_data = decoder.decode_bits()[0]
+                            decoded_data = self.__decode_from_registers(decoder, configuration)
```
Comment (samson0v) [inline thingsboard_gateway/connectors/modbus/bytes_modbus_uplink_converter.py:66]: this have to be 
`decoded_data = self.decode_from_registers(decoder, configuration)`

Comment (samson0v) [reply thingsboard_gateway/connectors/modbus/bytes_modbus_uplink_converter.py:66]: decode_from_registers is not a private method

File: thingsboard_gateway/connectors/modbus/bytes_modbus_uplink_converter.py (hunk 1 above)
Comment (samson0v) [inline thingsboard_gateway/connectors/modbus/bytes_modbus_uplink_converter.py:56]: add space `[1, 2]`

File: thingsboard_gateway/connectors/modbus/bytes_modbus_uplink_converter.py [hunk 3]
```diff
@@ -62,8 +62,9 @@ def convert(self, config, data):
                                 decoder = BinaryPayloadDecoder.fromCoils(coils, wordorder=word_endian_order)
                             assert decoder is not None
 
-                            decoded_data = decoder.decode_bits()[0]
-                        elif configuration["functionCode"] in [2, 3, 4]:
+                            #decoded_data = decoder.decode_bits()[0]
```
Comment (samson0v) [inline thingsboard_gateway/connectors/modbus/bytes_modbus_uplink_converter.py:65]: delete unused comments in all your modified files

**Comment by samson0v** [comment]
@zhufeng88, please, resolve all conversations above and we will merge your PR
